cmd/buildkitd/main_unix.go

```diff
@@ -63,3 +75,10 @@ func getLocalListener(listenerPath string) (net.Listener, error) {
 	}
 	return l, nil
 }
+
+func runtimeDir() string {
+	if xdg := os.Getenv("XDG_RUNTIME_DIR"); xdg != "" {
+		return xdg
+	}
+	return fmt.Sprintf("/run/user/%d", os.Getuid())
```

Comment:
When running as the root this has to be `/var`.
(The logic should be just same as `buildkitd.sock`)